File: crates/nil-payload/src/round.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetRoundRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SetPlayerReadyRequest {
  pub world: WorldId,
  pub is_ready: bool,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StartRoundRequest {
  pub world: WorldId,
}

File: app/src/types/request/round.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { WorldId } from '@/types/core/world';

export interface GetRoundRequest {
  readonly world: WorldId;
}

export interface SetPlayerReadyRequest {
  readonly world: WorldId;
  readonly isReady: boolean;
}

export interface StartRoundRequest {
  readonly world: WorldId;
}
